homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_zone_control_zone.py

```diff
@@ -60,48 +116,83 @@ def is_using_derogated_temperature_fallback(self) -> bool:
+        ) is not None:
+            return OVERKIZ_TO_HVAC_ACTION[cast(str, state)]
+
+        return HVACAction.OFF
 
     @property
-    def hvac_mode(self) -> HVACMode:
-        """Return hvac operation ie. heat, cool, dry, off mode."""
+    def is_zone_control_auto_switch_active(self) -> bool:
+        """Check if auto mode is available and active on the ZoneControl."""
 
-        if self.is_using_derogated_temperature_fallback:
-            return super().hvac_mode
+        if (
+            state := self.zone_control_executor.select_state(
+                OverkizState.CORE_HEATING_COOLING_AUTO_SWITCH
+            )
+        ) is not None:
+            return cast(str, state) == OverkizCommandParam.ON
+
+        return False
 
-        zone_control_hvac_mode = self.zone_control_hvac_mode
+    @property
+    def hvac_action(self) -> HVACAction | None:
+        """Return the current running hvac operation."""
 
-        # Should be same, because either thermostat or this integration change both.
-        on_off_state = cast(
+        # When ZoneControl action is heating/cooling but Zone is stopped, means the zone is idle.
+        if (
+            hvac_action := self.zone_control_hvac_action
+        ) in HVAC_ACTION_TO_OVERKIZ_PROFILE_STATE and cast(
             str,
             self.executor.select_state(
-                OverkizState.CORE_COOLING_ON_OFF
-                if zone_control_hvac_mode == HVACMode.COOL
-                else OverkizState.CORE_HEATING_ON_OFF
+                HVAC_ACTION_TO_OVERKIZ_PROFILE_STATE[hvac_action]
             ),
-        )
+        ) == OverkizCommandParam.STOP:
+            return HVACAction.IDLE
 
-        # Device is Stopped, it means the air flux is flowing but its venting door is closed.
-        if on_off_state == OverkizCommandParam.OFF:
-            hvac_mode = HVACMode.OFF
+        return hvac_action
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool, dry, off mode."""
+
+        if self.is_using_derogated_temperature_fallback:
+            return super().hvac_mode
+
+        zone_control_hvac_mode = HVAC_ACTION_TO_HVAC_MODES[
+            zone_control_hvac_action := self.zone_control_hvac_action
+        ]
+
+        # Selectable mode is kept in synced with the Zone Control current mode.
+        # When auto switching is enabled on Zone Control, we can use the HEAT_COOL mode.
+        if self.is_zone_control_auto_switch_active:
+            selectable_hvac_mode = HVACMode.HEAT_COOL
         else:
-            hvac_mode = zone_control_hvac_mode
+            selectable_hvac_mode = zone_control_hvac_mode
 
-        # It helps keep it consistent with the Zone Control, within the interface.
-        if self._attr_hvac_modes != [zone_control_hvac_mode, HVACMode.OFF]:
-            self._attr_hvac_modes = [zone_control_hvac_mode, HVACMode.OFF]
-            self.async_write_ha_state()
+        self.sync_attrs_from_mode(selectable_hvac_mode)
```

Comment:
@MartinHjelmare thank you for this feedback, I was crying when I wrote it (main reason why it's a Draft PR).
With the HEAT_COOL (for every main device mode), this may not be needed anymore.

We however loose a bit of control on the device because:
if I enable HEAT_COOL (on subsidiaries devices)for other mode than AUTO (on main device), it seams cooling temperature needs to be strictly higher than heating temperature => so it means, if I want to set heating to 21 and cooling to 20 (for whatever reason), and switching between HEATING and COOLING with an automation, I can not do that.
However it may still makes some sense, because then I'll be able to always choose cooling AND heating temperature whatever the mode (with the only limitation being the cooling_temp > heating_temp)

I'll go this way, I'll handle that asap.